src/main/java/games/strategy/triplea/ui/BattleDisplay.java

```diff
@@ -425,13 +425,18 @@ private Territory getRetreatInternal(final String message, final Collection<Terr
       // if you have eliminated the impossible, whatever remains, no matter
       // how improbable, must be the truth
       // retreat
-      final RetreatComponent comp = new RetreatComponent(possible);
-      final int option = JOptionPane.showConfirmDialog(BattleDisplay.this, comp, message,
-          JOptionPane.OK_CANCEL_OPTION, JOptionPane.PLAIN_MESSAGE, null);
-      if (option == JOptionPane.OK_OPTION) {
-        if (comp.getSelection() != null) {
-          retreatTo[0] = comp.getSelection();
-          latch.countDown();
+      if (possible.size() == 1) {
+        retreatTo[0] = possible.iterator().next();
+        latch.countDown();
```

Comment:
Surely that is the responsibility of the attacker though.

I suppose we could move the adding the battle territory when there are no retreat territories to after the prompt and look for size() == 0 but then I would still ask: Why?

Regarding your suggestion of the newly conquered province, the only way that is possible is if you blitzed through it is it not? In that case, retreating to the newly conquered territory would be completely valid.